reference/generators.rst

```diff
@@ -43,3 +43,5 @@ Available generators:
    generators/json
    generators/premake
    generators/make
+   generators/deploy
+   generators/custom
```

Comment:
I still think we should put this out of the list (in a line or in a tip/note):

```suggestion

If none of these generators fits your needs, you can create your own
:ref:`custom generator <custom_generator>`.
```

The only caveat is that we will need the `:orphan:` keyword inside `reference/generators/custom.rst` to avoid the rst error.

Reply:
Still don't like the idea of having the section so hidden, but I have taken your suggestion and we can change it in the future. Thanks for the helpful reviews

Reply:
I would also add an entry, linking to this, in ``Extending Conan``. This is clearly a point of extension and should be summarized there. Can be a short summary: purpose, goal, brief description and short recipe and link to main page

Reply:
I think in that case we should move the how-to to the ``Extending Conan`` section. Otherwise, there would be too many entries for generators with bits of information around the place. I will create an issue apart so this PR can be merged